setup.py

```diff
@@ -626,6 +630,34 @@ def parse_version(version_str):
     return version
 
 
+def is_mx_mkldnn():
+    try:
+        from mxnet import runtime
+        features = runtime.Features()
+        return features.is_enabled('MKLDNN')
+    except Exception:
+        msg = 'INFO: Cannot detect if MKLDNN is enabled in MXNet. Please \
+            set MXNET_USE_MKLDNN=1 if MKLDNN is enabled in your MXNet build.'
+        if 'linux' not in sys.platform:
+            # MKLDNN is only enabled by default in MXNet Linux build. Return 
+            # False by default for non-linux build but still allow users to 
+            # enable it by using MXNET_USE_MKLDNN env variable. 
+            print(msg)
+            return os.environ.get('MXNET_USE_MKLDNN', '0') == '1'
+        else:
+            try:
+                import mxnet as mx
+                mx_libs = mx.libinfo.find_lib_path()
+                for mx_lib in mx_libs:
+                    output = subprocess.check_output(['readelf', '-d', mx_lib])
+                    if 'mkldnn' in str(output):
+                        return True
+                    return False
+            except Exception:
+                print(msg)
+                return os.environ.get('MXNET_USE_MKLDNN', '0') == '1'
```

Comment:
Should we just return false in this case?

Reply:
we want to give users an option to set it if we fail to automatically detect it, just like for non-linux build.

Reply:
If user sets env variable to 1 but there is no libraries available, this will still cause the segmentation fault, right?

Reply:
When we are here, it means that we failed to automatically detect if mkldnn is enabled in the mxnet library, we do not know if mkldnn is there or not. So we give users the option to set it by themselves.

Reply:
Discussed offline. This seems not an issue,  but leave option to user.